e2e_test/batch/basic/union.slt.part

```diff
@@ -69,9 +69,55 @@ NULL
 statement error
 select null union all select null select union 1
 
+query II
+select * from t1 union all corresponding select * from t2 order by v1
+----
+1 3
+1 3
+
+query II
+select * from t1 union corresponding select v4, v3 as v1 from t2 order by v1
+----
+1 3
+2 3
+
+query II
+select * from t1 union all corresponding by (v4, v1) select * from t2
+----
+3 1
+3 1
+
+query II
+select * from t1 union corresponding by (v4) select * from t2
+----
+3
+
+statement error
+select * from t1 union corresponding by (vxx) select * from t2
+----
+db error: ERROR: Failed to run the query
+
+Caused by:
+  Invalid input syntax: Every column name in the corresponding column list shall be a column name of both left and right side query in a UNION operation. Column name `vxx` in the corresponding column list is not found in a query column list: ("v1", "v2", "v4").
+
+
+statement ok
+create table txx (vxx int);
+
+statement error
+select * from t1 union corresponding select * from txx
+----
+db error: ERROR: Failed to run the query
+
+Caused by:
+  Invalid input syntax: When CORRESPONDING is specified, at least one column of the left side shall have a column name that is the column name of some column of the right side in a UNION operation. Left side query column list: ("v1", "v2", "v4"). Right side query column list: ("vxx").
```

Comment:
@xxchan should we change this too?

Reply:
You can decide it :)

Reply:
I'll keep this considering it can be fairly difficult to identify the faulty query without any information.